x-pack/plugins/infra/common/alerting/metrics/alert_link.ts

```diff
@@ -24,8 +24,8 @@ export const getInventoryViewInAppUrl = (
     };
     // We always pick the first criteria metric for the URL
     const criteriaMetric = fields[`${ALERT_RULE_PARAMETERS}.criteria.metric`][0];
-    const criteriaCustomMetricId = fields[`${ALERT_RULE_PARAMETERS}.criteria.customMetric.id`][0];
-    if (criteriaCustomMetricId !== 'alert-custom-metric') {
+    if (criteriaMetric === 'custom') {
+      const criteriaCustomMetricId = fields[`${ALERT_RULE_PARAMETERS}.criteria.customMetric.id`][0];
```

Comment:
@simianhacker This change looks reasonable to me. I am just wondering how it used to work in the first place. Here's the PR that introduced this change https://github.com/elastic/kibana/pull/113553/files#diff-173c0f4e56c53ef08415cbd7e2a1773b2bd84596e5c366be4fb55507d33265fdR2. I'll take a closer look

Reply:
I'm not sure why it was written that way, probably just a misunderstanding of how the custom metric stuff works.